src/test/scala/zio/config/testsupport/TestSupport.scala

```diff
@@ -0,0 +1,89 @@
+package zio.config.testsupport
+
+import org.scalacheck.{ Arbitrary, Gen }
+import zio.{ DefaultRuntime, ZIO }
+
+final class TestSupportOps[A](val actual: A) {
+  def shouldBe(expected: A): Boolean = {
+    val result = expected == actual
+    if (!result) {
+      println(s"       => FAIL: expected[$expected]")
+      println(s"                  actual[$actual]")
+    }
+    result
```

Comment:
One thing that you may be interested in trying out is making `shouldBe`  returning ZIO ?
I haven't investigated zoo-test. But even we are not using it, it will be a nice one to have at this stage with scalaCheck.  
Something along the lines of `read(config).test(mapSource(xyz)) shouldBe ("something")` .